app/src/main/java/app/revanced/integrations/youtube/settings/preference/ExternalPlaylistDownloaderPreference.java

```diff
@@ -0,0 +1,162 @@
+package app.revanced.integrations.youtube.settings.preference;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.preference.Preference;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.AttributeSet;
+import android.util.TypedValue;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.TableLayout;
+import android.widget.TableRow;
+import app.revanced.integrations.shared.settings.StringSetting;
+import app.revanced.integrations.shared.utils.ResourceUtils;
+import app.revanced.integrations.shared.utils.Utils;
+import app.revanced.integrations.youtube.settings.Settings;
+import app.revanced.integrations.youtube.utils.ExtendedUtils;
+
+import java.util.Arrays;
+
+import static app.revanced.integrations.shared.utils.StringRef.str;
+
+/**
+ * @noinspection all
+ */
+public class ExternalPlaylistDownloaderPreference extends Preference implements Preference.OnPreferenceClickListener {
+
+    private static final StringSetting settings = Settings.EXTERNAL_PLAYLIST_DOWNLOADER_BUTTON;
+    private static final String[] mEntries = ResourceUtils.getStringArray("revanced_external_downloader_label");
```

Comment:
As of now, the only working external playlist downloader is `YTDLnis`.
You can remove any other presets.